backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/process/M_ReceiptSchedule_ChangeDatePromised_OverrideAndPOReference.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * #%L
+ * de.metas.swat.base
+ * %%
+ * Copyright (C) 2025 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.inoutcandidate.process;
+
+import de.metas.i18n.AdMessageKey;
+import de.metas.inoutcandidate.api.IReceiptScheduleBL;
+import de.metas.inoutcandidate.api.IReceiptScheduleDAO;
+import de.metas.inoutcandidate.model.I_M_ReceiptSchedule;
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.IProcessPreconditionsContext;
+import de.metas.process.JavaProcess;
+import de.metas.process.Param;
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+
+import java.time.LocalDate;
+
+public class M_ReceiptSchedule_ChangeDatePromised_OverrideAndPOReference extends JavaProcess implements IProcessPrecondition
+{
+	private final IReceiptScheduleDAO receiptScheduleDAO = Services.get(IReceiptScheduleDAO.class);
+	private final IReceiptScheduleBL receiptScheduleBL = Services.get(IReceiptScheduleBL.class);
+
+	private static final AdMessageKey MSG_NO_UNPROCESSED_LINES = AdMessageKey.of("receiptschedule.noUnprocessedLines");
+
+	@Param(parameterName = I_M_ReceiptSchedule.COLUMNNAME_DatePromised_Override)
+	private LocalDate datePromisedOverride;
+
+	@Param(parameterName = I_M_ReceiptSchedule.COLUMNNAME_POReference)
+	private String poReference;
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable(@NonNull final IProcessPreconditionsContext context)
+	{
+		if (context.isNoSelection())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+
+		final boolean someSchedsAreStillNotProcessed = context.streamSelectedModels(I_M_ReceiptSchedule.class)
+				.anyMatch(receiptSchedule -> !receiptScheduleBL.isClosed(receiptSchedule));
+
+		if (!someSchedsAreStillNotProcessed)
+		{
+			return ProcessPreconditionsResolution.reject(msgBL.getTranslatableMsgText(MSG_NO_UNPROCESSED_LINES));
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Override
+	protected void prepare()
+	{
+		final int selectionCount = receiptScheduleDAO
```

Comment:
IMHO u can merge the whole prepare() into doIt(). there is no point having prepare here.